ingress/loadbalancer.go

```diff
@@ -17,19 +37,181 @@ type LoadBalancerEntry struct {
 	ServicePort int32
 }
 
-type noopLoadBalancer struct {
+// Nginx implementation
+type nginxLoadBalancer struct {
+	binary    string
+	confDir   string
+	port      int
+	cmd       *exec.Cmd
+	signaller Signaller
 }
 
-func (lb *noopLoadBalancer) Update(entries []LoadBalancerEntry) error {
-	log.Debugf("Updating loadbalancer for %v", entries)
-	return nil
+// Used for generating nginx cofnig
+type loadBalancerTemplate struct {
+	Port       int
+	Entries    []LoadBalancerEntry
+	WorkingDir string
+}
+
+// DefaultSignaller sends os signals
+type DefaultSignaller struct {
+}
+
+// Sigquit sends a SIGQUIT to the process
+func (s *DefaultSignaller) Sigquit(p *os.Process) error {
+	return p.Signal(syscall.SIGQUIT)
+}
+
+// Sighup sends a SIGHUP to the process
+func (s *DefaultSignaller) Sighup(p *os.Process) error {
+	return p.Signal(syscall.SIGHUP)
 }
 
 // NewLB creates a new LoadBalancer
-func NewLB() LoadBalancer {
-	return &noopLoadBalancer{}
+func NewLB(nginxLocation, nginxConfDir string, port int, signaller Signaller) LoadBalancer {
+	log.Infof("Creating nginx with %s binary in %s", nginxLocation, nginxConfDir)
+	return &nginxLoadBalancer{
+		binary:    nginxLocation,
+		confDir:   nginxConfDir,
+		port:      port,
+		signaller: signaller}
+}
+
+func (lb *nginxLoadBalancer) Start() error {
+	// Write out initial configuration
+	lb.update(LoadBalancerUpdate{Entries: []LoadBalancerEntry{}})
```

Comment:
Hrmmmmmmmmmmmmmm

Indeed. Perhaps we should just include a default configuration in the container that is picked on initial start. I will create a separate task for this and work on it after this is PRed